### PR TITLE
Add version to EnvironmentContext

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs [hunk 1]
```diff
@@ -40,6 +41,25 @@ public static void SetOutputDirectory(string dir)
...
+            get
+            {
+                var version = _version;
+                if (string.IsNullOrEmpty(version))
+                {
+                    version = typeof(EnvironmentContext).Assembly.GetName().Version.ToString();
+                }
+
+                _version = version;
+                return _version;
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs]: temp variable `version` is not needed?

File: src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs [hunk 2]
```diff
@@ -40,6 +41,25 @@ public static void SetOutputDirectory(string dir)
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static IFileAbstractLayer FileAbstractLayerImpl { get; set; }
 
+        public static string Version
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs]: >Version [](start = 29, length = 7)

What about `AssemblyVersion`? Version in other place is moniker...

File: src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs (hunk 1 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs]: for thread safety

File: src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs [hunk 3]
```diff
@@ -41,25 +41,15 @@ public static void SetOutputDirectory(string dir)
...
-            {
-                _version = value;
-            }
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void SetVersion(string version)
+        {
+            _version = version;
         }
 
+
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Plugins/EnvironmentContext.cs]: minor: remove empty line

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**